src/ipips/ipip-0431.md

```diff
@@ -0,0 +1,125 @@
+
+The metadata format is DAG-CBOR and open to extension, allowing standardized
+userland experimentation similar to the Extensible Data field from IPNS V2.
+
+See [CAR `meta` (content type parameter)](/http-gateways/trustless-gateway/#car-meta-content-type-parameter)
+in Trustless Gateway specification for more details.
+
+## Design rationale
+
+The proposal introduces a minimal change allowing Gateways and retrieval clients to explicitly opt
+into receiving additional metadata block at the end of the CAR response stream.
+
+The metadata block is designed to be very flexible and able to support new use-cases that may arise
+in the future.
+
+### User benefit
+
+- Clients of trustless gateways can use the fields from the metadata as an attestation that they
+performed the retrieval from the given server.
+
+- The `len` field in the metadata block allows clients to verify whether they received all CAR
+bytes, which provides a backward-compatible solution for the [CARv1 streaming problem](https://github.com/ipfs/specs/pull/332) until new CAR version is introduced.
+
+### Compatibility
+
+The new feature requires clients to explicitly ask the server to include the extra block via `Accept` header,
+therefore the change is fully backwards-compatible for all existing gateway clients.
+
+Gateways receiving requests for the CAR content type can ignore the `meta` parameter they don't
+support and return back a response with one of the CAR content types they support. This makes the
+proposed change backwards-compatible for existing gateways too.
+
+
+### Security
+
+The proposed specification change does not introduce any negative security implications.
+
+### Alternatives
+
+#### HTTP Trailers
+
+Instead of adding a new content type argument, we were considering sending the additional metadata
+in HTTP response trailers. Unfortunately, HTTP trailers are not widely supported by the ecosystem.
+Nginx proxy module discards them, [browser `Fetch API` does not allow JS clients to access trailer
+headers](https://github.com/mdn/browser-compat-data/issues/14703), neither does the Rust `reqwest` client.
+
+#### New Content-Type
+
+We could introduce new `Content-Type: application/vnd.ipld.car-stream` and
+create a specification of its wire format that wraps CARv1 and includes
+additional DAG-CBOR  manifest at the end. It would be effectively the same CAR
+byte stream, but with different `Content-Type`.
+
+Upside of this solution:
+
+- does not require registering new codec, or sniffing  the last DAG-CBOR block
+
+Downsides of this solution:
+
+- maintenance cost, requires duplicating of all CAR-related tests and features
+- ecosystem opportunity cost, in creating new content type, we increase
+  cognitive overhead for everyone working with IPFS over HTTP
+- no backward-compatible interop with existing tools and gateways that only
+  speak `application/vnd.ipld.car`
+- distracts us away from working on things like large blocks and CARv3
```

Comment:
:information_source:  added a new content type as  "Plan B" to "Alternatives" here, with some initial pros/cons.